### PR TITLE
MM-56324 Channel Header UI Update

### PR DESCRIPTION
**Pull request by matthewbirtch**
#### Summary
This PR updates the channel header UI to simplify to a single row - inline with the design work done for the channel bookmarks bar.

The following changes are included:
- Shorter height of the channel header
- Elements in channel header have been re-arranged and re-ordered to be on a single row
- Shorter height of the RHS header to align
- Tweaks on the apps sidebar padding to align
- Minor padding added to LHS header to align
- Uses new icon button styles/classes for icon buttons in channel header
- Minor updates to header popover styles/positioning
- "Add channel header" text only shows on hover of the header area
- DM header includes the user's avatar + status

#### Ticket Link
Jira ticket: https://mattermost.atlassian.net/browse/MM-56324

#### Screenshots
|  before  |  after  |
|----|----|
| image | image |
| image | image |
| image | image | 
| image | image |

#### Release Note

```release-note
Updated the channel header layout to reduce height and simplify the UI.
```

### CODE REVIEW

**Comment by esethna** [comment]
@matthewbirtch @asaadmahmood (copying comment over here since I posted it on the wrong PR)...

Here's the case I was mentioning earlier (ie narrow screen or portrait mode monitor) with RHS open. Here I cannot even see or access the channel header:

image

With a shorter channel name, the header pop-over feels really narrow if it shows at all:

image

**Comment by matthewbirtch** [comment]
@esethna I made some updates to better handle long channel names by truncating them when the extend beyond 60% of the width of the header content area. @asaadmahmood I may need your help to add a tooltip when it's being truncated (but only when needed)

**Review by asaadmahmood — APPROVED** [review]
LGTM

**Comment by matthewbirtch** [comment]
> @asaadmahmood I may need your help to add a tooltip when it's being truncated (but only when needed)

@asaadmahmood do you think you could help out with the tooltip on truncated long channel names?

File: webapp/channels/src/sass/layout/_headers.scss
Comment (abhijit-singh) [inline webapp/channels/src/sass/layout/_headers.scss:319]: gap: 4px;

Recommend adding 4px space between the Channel title menu item and the icon button. Currently looks odd in the hover and selected state. 0/5 on how to do it, this is just one way, can be done via margins also. 
image

**Review by abhijit-singh — APPROVED** [review]
Just one comment, but other than that, looks great!

File: webapp/channels/src/sass/layout/_headers.scss
Comment (matthewbirtch) [reply webapp/channels/src/sass/layout/_headers.scss:331]: Thanks @abhijit-singh I've updated this in the latest commit

**Review by calebroseland — APPROVED** [review]
LGTM, just one potential edge case when a custom status is set, at narrow viewport widths the channel header can become inaccessible.

CleanShot 2024-04-09 at 16 35 03
CleanShot 2024-04-09 at 16 34 24

**Comment by matthewbirtch** [comment]
> LGTM, just one potential edge case when a custom status is set, at narrow viewport widths the channel header can become inaccessible.

Thanks Caleb, I don't have an easy solution to rectify that as we have to hide _something_ when we get to that small of a width. We could potentially hide the custom status text below a certain viewport width (since there is already a tooltip on the custom status emojis). That's the only thing I could think of though. It wouldn't solve it perfectly. In some ways we have to live with some information being hidden in small windows in cases like this.

**Review by lindy65 — APPROVED** [review]
Thanks @matthewbirtch - looks good to me 👍

**Review by enahum — APPROVED** [review]

**Comment by matthewbirtch** [comment]
@esethna gentle nudge to have a look at the latest

**Review by esethna — APPROVED** [review]
Looks really good, great work on the responsive layout

**Comment by esethna** [comment]
@matthewbirtch sure v9.9 works for me and would give us more soak time on hub/community

**Comment by esethna** [comment]
@calebroseland just curious if/how merging this would impact the open bookmarks PR? Will it need significant changes to address conflicts?

**Comment by calebroseland** [comment]
> @calebroseland just curious if/how merging this would impact the open bookmarks PR? Will it need significant changes to address conflicts?

@esethna no, this shouldn’t conflict with #25889 much, if at all. Should be good to go from that perspective.

**Comment by matthewbirtch** [comment]
@esethna @calebroseland should we merge? I don't have authorization to do it, but it would be good to dogfood this.

**Comment by esethna** [comment]
Nothing blocking merge from my side @matthewbirtch

**Comment by matthewbirtch** [comment]
> Nothing blocking merge from my side @matthewbirtch

Thanks @esethna. Since I'm not able to merge, will need @calebroseland or @enahum to press the merge button here

**Pull request merged**
